packages/test/test-end-to-end-tests/src/test/generateSummary.spec.ts

```diff
@@ -0,0 +1,178 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import assert from "assert";
+import {
+    ContainerRuntimeFactoryWithDefaultDataStore,
+    DataObject,
+    DataObjectFactory,
+} from "@fluidframework/aqueduct";
+import { TelemetryNullLogger } from "@fluidframework/common-utils";
+import { ITelemetryBaseEvent } from "@fluidframework/common-definitions";
+import { IContainer } from "@fluidframework/container-definitions";
+import { ISummaryConfiguration } from "@fluidframework/protocol-definitions";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import { ITestObjectProvider } from "@fluidframework/test-utils";
+import { MockLogger } from "@fluidframework/test-runtime-utils";
+import { describeNoCompat } from "@fluidframework/test-version-utils";
+import { IContainerRuntimeOptions, SummaryCollection } from "@fluidframework/container-runtime";
+import { flattenRuntimeOptions } from "./flattenRuntimeOptions";
+
+const _global: any = global;
+class TestDataObject extends DataObject {
+    public get _root() {
+        return this.root;
+    }
+
+    public get _runtime() {
+        return this.runtime;
+    }
+
+    public get _context() {
+        return this.context;
+    }
+}
+
+function getEvent(events: ITelemetryBaseEvent[], sequenceNumber: number): any {
```

Comment:
This function can move inside the describe and then you don't need to pass in the `events` param. The function will have access to it.

Reply:
The return type should be `ITelemetryBaseEvent | undefined`. Then at the call site, you can assert that its not undefined. Or you can make the type `ITelemetryBaseEvent` and assert that you find the event inside this function itself.

Reply:
This function should be called `getGenerateSummaryEvent` or something similar since it only looks for that event.